### PR TITLE
feat: Added the option to declare a specific resource type to be removed last

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- Added the option to declare a specific resource type to be removed last
- This is required if a resource like a subscription should be removed after everything else is removed. This is done in preparation for the LZ-Accelerator 

## Pipeline Reference

| Pipeline |
| -------- |
|     [avm.res.key-vault.vault](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.res.key-vault.vault.yml)     |

## Type of Change

- [x] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

### CODE REVIEW

File: avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1 [hunk 1]
```diff
@@ -95,6 +95,10 @@ function Initialize-DeploymentRemoval {
             'Microsoft.Resources/resourceGroups'
         )
 
+        $removeLastSequence = @(
```
Comment (eriqua) [inline avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1:98]: Just to double check: If multiple elements are specified in this list, in which order will they be removed?

Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1:98]: Exactly. You can see this tested here: 
https://github.com/Azure/bicep-registry-modules/blob/f1012df886a4052b67dd87e385b87c4c18f8ce62/avm/utilities/tests/pipelines/Get-OrderedResourcesList.tests.ps1#L77-L144

(different maybe PR that I mentioned)

Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1:98]: Hence I was so keen on having a Pester test 😄

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**